pkg/scaling/executor/scale_jobs.go

```diff
@@ -273,18 +322,18 @@ type customScalingStrategy struct {
 	CustomScalingRunningJobPercentage *float64
 }
 
-func (s customScalingStrategy) GetEffectiveMaxScale(maxScale, runningJobCount, maxReplicaCount int64) int64 {
+func (s customScalingStrategy) GetEffectiveMaxScale(maxScale, runningJobCount, pendingJobCount, maxReplicaCount int64) int64 {
 	return min(maxScale-int64(*s.CustomScalingQueueLengthDeduction)-int64(float64(runningJobCount)*(*s.CustomScalingRunningJobPercentage)), maxReplicaCount)
 }
 
 type accurateScalingStrategy struct {
 }
 
-func (s accurateScalingStrategy) GetEffectiveMaxScale(maxScale, runningJobCount, maxReplicaCount int64) int64 {
+func (s accurateScalingStrategy) GetEffectiveMaxScale(maxScale, runningJobCount, pendingJobCount, maxReplicaCount int64) int64 {
 	if (maxScale + runningJobCount) > maxReplicaCount {
 		return maxReplicaCount - runningJobCount
 	}
-	return maxScale
+	return maxScale - pendingJobCount
```

Comment:
@zroubalik I don't think this is quite correct. i think pendingJobCount needs to be taken into account in the comparison to maxReplicaCount.

If we say
```
maxScale = 10
runningJobCount = 5
pendingJobCount = 5
maxReplicaCount = 20
```
we get `5`, which is correct.
`10 + 5 = 15 < 20 -=> 10 - 5 = 5`

However, if we say:
```
maxScale = 10
runningJobCount = 5
pendingJobCount = 5
maxReplicaCount = 10
```
then we get the return  `5`, even though `runningJobCount + pendingJobCount = 10`, so the maxReplicaCount is already hit.
`(10 + 5 = 15) > 10 -=> 10 - 5 = 5`

So this should instead be 
```
if (maxScale + runningJobCount) > maxReplicaCount {
     return maxReplicaCount - runningJobCount  - pendingJobCount
}
return maxScale - pendingJobCount
```
or to simplify
```
return min(maxReplicaCount - runningJobCount, maxScale) - pendingJobCount
```

Reference: https://github.com/kedacore/keda/pull/1227#issuecomment-726426728

Reply:
Hi @fjmacagno,

I think the problem here is the terminology, `runningJobCount` is in fact "jobs that are not finished", so in your example the 5 running jobs are the 5 pending jobs. so the result =5 is good, the runner can accept 5 other jobs.
If you subtract again `runningJobCount`, you will never scale to `maxReplicaCount`.

Reply:
Hi @fjmacagno I tend to share @thomas-lamure view on this. But I am happy to hear more from you, whether we need to change something, or change docs in some area. Thanks!